internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -3140,30 +3179,54 @@ func (cmi *ClusterManager) batchUpdateComponentRVState(msgBatch []*dcache.Compon
 				continue
 			}
 
+			//
+			// If the RV/MV combination is already present in the batch, we can ignore this update as it
+			// was already processed in the previous iteration. We do assert that the previous state
+			// is same as the new state requested, so that we don't end up with multiple updates for the same
+			// RV/MV combination in the batch.
+			//
+			if rvPrevState, ok := existing[mvName+rvName]; ok {
+				_ = rvPrevState
+				common.Assert(rvPrevState == rvNewState)
```

Comment:
we can also assert that currentState==rvNewState (prev update would have changed the state) or it's the case where currentState is offline and rvNewState is inband-offline (in which case the prev update would not have changed the state).

Reply:
We cannot assert `currentState==rvNewState`. For example, there are multiple duplicate entries in the batch to update rv0/mv0 to inband-offline from online state. In the first entry, we add it to the existing map. In the next entry, we ignore it since it was already processed in the previous duplicate entry. Here, currentState(online) is not equal to rvNewState(inband-offline).